src/main/java/minijava/ir/assembler/block/AssemblerFile.java

```diff
@@ -0,0 +1,126 @@
+package minijava.ir.assembler.block;
+
+import com.sun.jna.Platform;
+import java.util.*;
+import minijava.ir.NameMangler;
+import minijava.ir.assembler.GNUAssemblerConvertible;
+import org.jetbrains.annotations.NotNull;
+
+/** List of segments that are combined to an assembler file */
+public class AssemblerFile implements GNUAssemblerConvertible, Collection<Segment> {
```

Comment:
Yeah, but what are you actually encapsulating here? Looking at the above function signature makes it pretty clear what it's doing, while having some class with some tenish instance methods, most of them modifying some hidden state isn't clear at all.